integration/rust/tests/integration/test_mirror_with_sharding.rs

```diff
@@ -0,0 +1,725 @@
+use serial_test::serial;
+use sqlx::{Connection, PgConnection};
+use std::collections::HashSet;
+use std::process::{Child, Command};
+use std::time::Duration;
+use tokio_postgres::{Client, NoTls, SimpleQueryMessage};
+
+// Drop guard to ensure pgdog is always killed
+struct PgdogGuard {
+    child: Child,
+}
+
+impl Drop for PgdogGuard {
+    fn drop(&mut self) {
+        // Always try to kill the process when the guard is dropped
+        let _ = self.child.kill();
+    }
+}
+
+// Helper struct to hold test configuration
+struct MirrorTestConfig {
+    config_file: String,
+    users_file: String,
+    exposure: f32,
+}
+
+impl MirrorTestConfig {
+    fn new(test_name: &str, exposure: f32) -> Self {
+        Self {
+            config_file: format!("/tmp/mirror_test_{}.toml", test_name),
+            users_file: format!("/tmp/mirror_test_{}_users.toml", test_name),
+            exposure,
+        }
+    }
+
+    fn write_configs(&self) -> Result<(), std::io::Error> {
+        let config = format!(
+            r#"
+[general]
+openmetrics_port = 9090
```

Comment:
You sure these run? These integration tests require pgdog to be running already, this would conflict on the port.

Reply:
True! I keep adding them here as they come up: https://github.com/levkk/pgdog/blob/26816db293b03aaf7875e855d6f76c1813282c9a/pgdog/src/admin/set.rs#L55

Works for testing general settings as long as the test suite runs serially (hence `--test-threads=1`). For more advanced stuff, we should be able to load a new config dynamically, e.g. `SET CONFIG <config encoded as base64 or something easy to pass to a query>`. Just for making testing easier, I can't imagine this would be used in prod.